File: baekjoon/DFS&BFS/17204_죽음의 게임.py
import sys
input = sys.stdin.readline
n, k = list(map(int, input().split()))
arr = [int(input()) for _ in range(n)]


def solution():
    queue = arr[0]
    cnt = 1

    for _ in range(n):
        if queue == k:
            return cnt

        queue = arr[queue]
        cnt += 1

    return -1


print(solution())

File: baekjoon/DFS&BFS/17204_죽음의 게임.js
const fs = require('fs')
const [nm, ...input] = fs
  .readFileSync('../input.txt')
  .toString()
  .trim()
  .split('\n')
const [n, k] = nm.split(' ').map(Number)
const arr = input.map(Number)

function solution() {
  let queue = arr[0]
  let cnt = 1

  for (let i = 0; i < n; i++) {
    if (queue === k) {
      return cnt
    }
    queue = arr[queue]
    cnt += 1
  }

  return -1
}

console.log(solution())
